pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/queries/ResourceBasedQueriesTest.java

```diff
@@ -280,7 +281,13 @@ private static Map<String, QueryTestCase> getTestCases()
       if (testFileUrl != null && new File(testFileUrl.getFile()).exists()) {
         Map<String, QueryTestCase> testCases = MAPPER.readValue(new File(testFileUrl.getFile()),
             new TypeReference<Map<String, QueryTestCase>>() { });
-        // TODO: potential test case conflicts between files, address later by throwing during setUp if already exist.
+        {
+          HashSet<String> hashSet = new HashSet<>(testCaseMap.keySet());
+          hashSet.retainAll(testCases.keySet());
+          if (!hashSet.isEmpty()) {
+            throw new IllegalArgumentException("testCase already exist for the following names: " + hashSet);
+          }
+        }
```

Comment:
just realized that there was a test case name conflict and only the last loaded one were executed. this should fix it